cmd/timestamp-cli/app/verify.go

```diff
@@ -130,22 +131,27 @@ func verifyArtifactWithTSR(ts *timestamp.Timestamp) error {
 		return err
 	}
 
-	h := ts.HashAlgorithm.New()
-	b := make([]byte, h.Size())
+	return verifyHashedMessages(ts.HashAlgorithm.New(), ts.HashedMessage, bytes.NewReader(artifactBytes))
+}
 
-	r := bytes.NewReader(artifactBytes)
-	n, err := r.Read(b)
-	if err == io.EOF {
-		return err
-	}
+func verifyHashedMessages(hashAlg hash.Hash, hashedMessage []byte, artifactReader io.Reader) error {
+	h := hashAlg
 
-	_, err = h.Write(b[:n])
-	if err != nil {
-		return fmt.Errorf("Failed to create local message hash")
+	b := make([]byte, h.Size())
+	for {
+		n, err := artifactReader.Read(b)
+		if err == io.EOF {
+			break
+		}
+
+		_, err = h.Write(b[:n])
+		if err != nil {
+			return fmt.Errorf("failed to create hash %w", err)
+		}
```

Comment:
```suggestion
	if _, err := io.Copy(h, artifactReader); err != nil {
		return fmt.Errorf("failed to create hash %w", err)
	}
```

Reply:
h is an `io.Writer` so you can just let the `io` library do the buffering internally.